src/main/java/com/flansmod/client/model/RenderGun.java

```diff
@@ -119,332 +115,272 @@ public void renderOffHandGun(EntityPlayer player, ItemStack offHandItemStack) {
-						GL11.glRotatef(model.rotateGunHorizontal * reloadRotate, 0F, 1F, 0F);
-						GL11.glRotatef(model.tiltGun * reloadRotate, 1F, 0F, 0F);
-						GL11.glTranslatef(model.translateGun.x * reloadRotate, model.translateGun.y * reloadRotate,
-								model.translateGun.z * reloadRotate);
-						break;
-					}
-					default:
-						break;
-					}
-				}
+	private void renderMeleeMovement(GunType gunType, GunAnimations animations) {
+		Vector3f meleePos = gunType.meleePath.get(animations.meleeAnimationProgress);
+		Vector3f nextMeleePos = animations.meleeAnimationProgress + 1 < gunType.meleePath.size() ? gunType.meleePath.get(animations.meleeAnimationProgress + 1) : new Vector3f();
+		GL11.glTranslatef(meleePos.x + (nextMeleePos.x - meleePos.x) * smoothing, meleePos.y + (nextMeleePos.y - meleePos.y) * smoothing, meleePos.z + (nextMeleePos.z - meleePos.z) * smoothing);
+		Vector3f meleeAngles = gunType.meleePathAngles.get(animations.meleeAnimationProgress);
+		Vector3f nextMeleeAngles = animations.meleeAnimationProgress + 1 < gunType.meleePathAngles.size() ? gunType.meleePathAngles.get(animations.meleeAnimationProgress + 1) : new Vector3f();
+		GL11.glRotatef(meleeAngles.y + (nextMeleeAngles.y - meleeAngles.y) * smoothing, 0F, 1F, 0F);
+		GL11.glRotatef(meleeAngles.z + (nextMeleeAngles.z - meleeAngles.z) * smoothing, 0F, 0F, 1F);
+		GL11.glRotatef(meleeAngles.x + (nextMeleeAngles.x - meleeAngles.x) * smoothing, 1F, 0F, 0F);
+	}
+
+	private void renderReloadMovement(GunAnimations animations, ModelGun model, GunType gunType, ItemStack item, int flip) {
+		float reloadRotate;
+
+		EnumAnimationType anim = model.animationType;
+		if (gunType.getGrip(item) != null && gunType.getSecondaryFire(item))
+			anim = gunType.getGrip(item).model.secondaryAnimType;
+
+		// Calculate the amount of tilt required for the reloading animation
+		float effectiveReloadAnimationProgress = animations.lastReloadAnimationProgress
+				+ (animations.reloadAnimationProgress - animations.lastReloadAnimationProgress) * smoothing;
+		reloadRotate = 1F;
+		if (effectiveReloadAnimationProgress < model.tiltGunTime)
+			reloadRotate = effectiveReloadAnimationProgress / model.tiltGunTime;
+		if (effectiveReloadAnimationProgress > model.tiltGunTime + model.unloadClipTime
+				+ model.loadClipTime)
+			reloadRotate = 1F - (effectiveReloadAnimationProgress
+					- (model.tiltGunTime + model.unloadClipTime + model.loadClipTime))
+					/ model.untiltGunTime;
+
+		// Rotate/translate the GUN dependent on the animation type
+		switch (anim) {
+			case BOTTOM_CLIP:
+			case PISTOL_CLIP:
+			case SHOTGUN:
+			case END_LOADED: {
+				GL11.glRotatef(60F * reloadRotate, 0F, 0F, 1F);
+				GL11.glRotatef(30F * reloadRotate * flip, 1F, 0F, 0F);
+				GL11.glTranslatef(0.25F * reloadRotate, 0F, 0F);
 				break;
 			}
-			default:
+			//TODO Investigate if all this can be deleted
+			case CUSTOMBOTTOM_CLIP:
+			case CUSTOMPISTOL_CLIP:
+			case CUSTOMSHOTGUN:
```

Comment:
huh, where these just never implemented? / All have the same behaviour here?

Reply:
Yeah theyre basically all the same so reduced them to one case

Reply:
Oke